packages/application-shell/src/components/navbar/menu-items.tsx

```diff
@@ -191,6 +194,12 @@ const MenuGroup = (props: MenuGroupProps) => {
     props.level === 2 && props.isActive && !props.isExpanded;
   return (
     <ul
+      style={{
+        top: props.isNewNavigationEnabled
+          ? // calculating the submenu position based on the parent's top position and the scroll position
+            `calc(${props.topPosition}px - ${props.scrollTop}px)`
+          : undefined,
+      }}
```

Comment:
nit: perhaps we could use emotion's css here
```suggestion
css={
        props.isNewNavigationEnabled &&
        css`
          // calculating the submenu position based on the parent's top position and the scroll position
          top: calc(${props.topPosition}px - ${props.scrollTop}px);
        `
      }
```